### DISCUSSION TITLE
Tutorial to integrate a Primefaces template in a JSF project with Jakarta

### DISCUSSION THREAD

**Discussion by danielemaddaluno** [primefaces/community#742] [PrimeFaces Templates] [answered]
I am looking for a tutorial on how to integrate a template in a JSF project.
In particular I'm interested in integrating the Freya template.
Any help is appreciated.
Thank you in advance.

**Comment by Yazan-Abuawwad** [comment]
if you build native jsf just use 
1 -jdk 17
2- put  <classifier>jakarta</classifier> in primeface dependency in pom file 
3- change all javax to jakarta in all import package in classes
4- change all javax to jakarta in all XML file  config

**Reply by danielemaddaluno** [reply]
Thank you for your suggestions. I'm going to write a small tutorial down there.

**Comment by danielemaddaluno** [comment, marked as answer]
Since I haven't found a clear tutorial I'm writing it:

1. Start from this basic jsf+jakarta project: [primefaces-test](https://github.com/primefaces/primefaces-test/tree/jakarta)
2. Unzip you template `freya-5.0.0.zip`
3. Install the `freya-theme-5.0.0-jakarta.jar` as a local maven dependency running something like: 
`mvn install:install-file -Dfile=freya-theme-5.0.0-jakarta.jar -DgroupId=org.primefaces -DartifactId=freya -Dversion=5.0.0-jakarta -Dpackaging=jar`
4. Add the dependency to the `pom.xml` file:
```
<dependency>
     <groupId>org.primefaces</groupId>
     <artifactId>freya</artifactId>
     <version>5.0.0-jakarta</version>
 </dependency>
```
5. Copy the files from the freya-layout-5.0.0 folder as shown here (focus on `freya-layout` folder, `sass` folder and all green `*.xhtml` files):
![image](https://github.com/primefaces/community/assets/1886810/e8aae67c-8683-41a6-a062-22c05d1dc241)

6. Edit the `test.xhtml` as this example:
```
<ui:composition xmlns="http://www.w3.org/1999/xhtml"
                xmlns:h="http://java.sun.com/jsf/html"
                xmlns:f="http://java.sun.com/jsf/core"
                xmlns:ui="http://java.sun.com/jsf/facelets"
                xmlns:p="http://primefaces.org/ui"
                template="/WEB-INF/template.xhtml">

    <ui:define name="title">Test Page</ui:define>

    <ui:define name="content">
        Content goes here
    </ui:define>

</ui:composition>
```
7. Edit the menu.xhtml commenting out every outcome that does not exists. Don't comment the `<fr:menu widgetVar="FreyaMenuWidget">`.
8. Copy the `GuestPreferences.java`  file from the tag folder (look it inside of the unzipped `freya-5.0.0.zip`) and put it inside your java files changing all javax to jakarta, like this:
```
package org.primefaces.test;
import org.primefaces.PrimeFaces;

import jakarta.annotation.PostConstruct;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import jakarta.enterprise.context.SessionScoped;
import jakarta.inject.Named;

@Named
@SessionScoped
public class GuestPreferences implements Serializable {
    /** leave unchanged **/
}

```
8. To view correctly the images in the sidebar copy the `demo` folder from the tag folder to your `webapp/resources` folder.
9. Edit the `web.xml` file and add this:
```
<context-param>
    <param-name>primefaces.THEME</param-name>
    <param-value>freya-#{guestPreferences.theme}</param-value>
</context-param>
```
10. That's it

**Comment by mertsincan** [comment]
Thanks @danielemaddaluno! In addition, these videos can help you to integrate layouts into your project;
https://www.dropbox.com/s/gat9a4ym373miyv/import_Barcelona_Theme_into_a_project.mp4?dl=0
https://www.dropbox.com/s/2p74g63wldbover/import_Ultima_Theme_into_a_project.mp4?dl=0

Best Regards,

**Reply by danielemaddaluno** [reply]
Thank you very much for your addition.
Anyway finally I managed to integrate it. I just needed some small directions, it's very nice and works perfectly.
Anyway something simple for beginners should be added to the documentation, I suggest...
